sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -82,7 +81,14 @@ private object DateTimeFormatterHelper {
 
   def buildFormatter(pattern: String, locale: Locale): DateTimeFormatter = {
     val builder = createBuilder().appendPattern(pattern)
-    toFormatter(builder, locale)
+    if (pattern.contains("y")) {
```

Comment:
Only contains 'y' might be overkill? Should we skip for the cases that 'y' in the escape for text? e.g `'year:' uuuu HH mm`

Reply:
good catch! we should use regex or write custom parsing code.

Reply:
Another ways is to get access to the following list in `DateTimeFormatterBuilder`:
```java
private final List<DateTimePrinterParser> printerParsers = new ArrayList<>();
```
via reflection, and find `NumberPrinterParser` with `field` == `ChronoField.YEAR_OF_ERA`.

Reply:
this doesn't work, as the type `DateTimePrinterParser` is not accessible outside.